pyogrio/tests/test_raw_io.py

```diff
@@ -456,9 +458,11 @@ def assert_equal_result(result1, result2):
     assert all([np.array_equal(f1, f2) for f1, f2 in zip(field_data1, field_data2)])
 
 
+@pytest.mark.filterwarnings("ignore:File /vsimem:RuntimeWarning")  # TODO
```

Comment:
When reading from bytes / file-like objects, we create a vsimem temporary in-memory file, but because we don't know the file type at that moment, we get a warning like *"File /vsimem/5332805636f74498b74520a2c132082b has GPKG application_id, but non conformant file extension"* 
Ideally we would solve that (since the user cannot do anything about that warning, since this name is auto-generated), but not sure how. We don't know the type of file when it's a file-like object, and it's only GDAL that infers it based on the content of the file. Maybe we should open an issue about upstream in GDAL to relax this constraint for vsimem files.